runtime/src/main/java/org/corfudb/runtime/collections/TxBuilder.java

```diff
@@ -0,0 +1,270 @@
+        });
+        return this;
+    }
+
+    /**
+     * Touches the specified key without mutating the version of the record.
+     * This provides read after write conflict semantics.
+     *
+     * @param tableName Table name to perform the touch on.
+     * @param key       Key to touch.
+     * @param <K>       Type of Key.
+     * @param <V>       Type of Value.
+     * @return TxBuilder instance.
+     */
+    @Nonnull
+    public <K extends Message, V extends Message, M extends Message>
+    TxBuilder touch(@Nonnull final String tableName,
+                    @Nonnull final K key) {
+        Table<K, V, M> table = getTable(tableName);
+        validateNamespace(table);
+        operations.add(() -> {
+            //TODO: Validate the get is executed.
+            CorfuRecord<V, M> record = table.get(key);
+        });
+        return this;
+    }
+
+    /**
+     * Deletes the specified key.
+     *
+     * @param tableName Table name to perform the delete on.
+     * @param key       Key.
+     * @param <K>       Type of Key.
+     * @param <V>       Type of Value.
+     * @return TxBuilder instance.
+     */
+    @Nonnull
+    public <K extends Message, V extends Message, M extends Message>
+    TxBuilder delete(@Nonnull final String tableName,
+                     @Nonnull final K key) {
+        Table<K, V, M> table = getTable(tableName);
+        validateNamespace(table);
+        operations.add(() -> table.delete(key));
+        return this;
+    }
+
+    private void txBegin() {
+        Transaction.TransactionBuilder transactionBuilder = this.objectsView
+                .TXBuild()
+                .type(TransactionType.OPTIMISTIC);
+        if (timestamp != null) {
+            transactionBuilder.snapshot(new Token(timestamp.getEpoch(), timestamp.getSequence()));
+        }
+        transactionBuilder.build().begin();
+    }
+
+    private void txEnd() {
+        this.objectsView.TXEnd();
+    }
+
+    /**
+     * Commit the transaction.
+     * The commit call begins a Corfu transaction at the latest timestamp, applies all the updates and then
+     * ends the Corfu transaction.
+     * The commit returns successfully if the transaction was committed.
+     * Otherwise this throws a TransactionAbortedException.
+     */
+    public void commit() {
+        commit(null);
+    }
+
+    /**
+     * Commit the transaction.
+     * The commit call begins a Corfu transaction at the specified snapshot or at the latest snapshot
+     * if no snapshot is specified, applies all the updates and then ends the Corfu transaction.
+     * The commit returns successfully if the transaction was committed.
+     * Otherwise this throws a TransactionAbortedException.
+     *
+     * @param timestamp Timestamp to commit the transaction on.
+     */
+    public void commit(final Timestamp timestamp) {
+        this.timestamp = timestamp;
+        try {
+            txBegin();
+            operations.forEach(Runnable::run);
```

Comment:
it's not async then what's the point for Runnable? Please provide your own interface for that.
Would be great if your `Runnable` could return `Result` class, then you can get rid of `finally` section (I mean you can simplify it)